services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/oauth/OpenIDConnectManagerImpl.java

```diff
@@ -205,14 +208,15 @@ public boolean hasUserGrantedConsent(UserInfo userInfo, OIDCAuthorizationRequest
 	public OAuthAuthorizationResponse authorizeClient(UserInfo userInfo,
 			OIDCAuthorizationRequest authorizationRequest) {
 		if (AuthorizationUtils.isUserAnonymous(userInfo)) {
+			// Perhaps this should be an OIDC Error Code: https://openid.net/specs/openid-connect-core-1_0.html#AuthError
 			throw new UnauthorizedException("Anonymous users may not provide access to OAuth clients.");
```

Comment:
I didn't change this error message because I am not sure what the proper error code should be (of the error codes in the OIDC section in the commented URL).

Reply:
login_required